ci/run-tests.sh

```diff
@@ -2,12 +2,13 @@
 
 cargo test \
     --features "test-programs/test_programs" \
+    --features "test-programs/test_programs_http" \
```

Comment:
We don't actually use this feature anymore, and its an oversight that it is still present in `test-programs/Cargo.toml`. (Once the component model was needed to build the wasi tests, there was no reason to gate the component-model feature on just http, so one feature for building all tests is sufficient.)

Reply:
removed.